myia/pipeline/steps.py

```diff
@@ -595,7 +603,14 @@ def convert_result(self, arg, orig_t, vm_t: AbstractScalar, backend):
 
 @overload  # noqa: F811
 def convert_result(self, arg, orig_t, vm_t: AbstractArray, backend):
-    return backend.to_numpy(arg)
+    a = backend.to_numpy(arg)
+    if isinstance(backend, NumpyChecker):
+        a = ArrayWrapper(
+            a,
+            dtype.type_to_np_dtype(orig_t.element.dtype()),
+            orig_t.values[SHAPE]
+            )
```

Comment:
This is the wrong check: you only want to return an `ArrayWrapper` when the `return_backend` option is True. This means that `return_backend` should be checked here. You could add it as an extra parameter to `convert_result`, which is called from `myia.pipeline.steps.Wrap`, or maybe tuck it in the `backend` object.